packages/babel-traverse/src/path/evaluation.ts

```diff
@@ -391,6 +391,25 @@ function _evaluate(path: NodePath, state) {
     }
   }
 
+  if (path.isMemberExpression()) {
+    const property = path.get("property");
+    const object = path.get("object");
+    let key: any;
+
+    if (path.node.computed) {
+      key = evaluateCached(property, state);
+      if (!state.confident) return deopt(property, state);
+    } else if (property.isIdentifier()) {
+      key = property.node.name;
+    } else {
+      return deopt(property, state);
+    }
+
+    const value = evaluateCached(object, state);
+    if (state.confident) return value[key];
```

Comment:
The `evaluate` method is meant to evaluate static code. It is not an alternative for `eval`, instead the evaluation result is always a string or number.

However this is not the case for general member expression. For example if `key` is `"constructor"`, `value[key]` will be a constructor function implemented by current Node.js running Babel.

Reply:
Would this be solvable by using `hasOwnProperty`, or do you see this as a dead end?